webui/react/src/hooks/useSettings.ts

```diff
@@ -246,53 +253,68 @@ const useSettings = <T>(config: SettingsConfig<T>): UseSettingsReturn<T> => {
         }
       }
     },
-    [user?.id, config.storagePath, settings],
+    [config.storagePath, loadableUser],
   );
 
   const resetSettings = useCallback(
-    async (settingsArray?: string[]) => {
-      if (!settings) return;
+    (settingsArray?: string[]) => {
+      const user = Loadable.match(loadableUser.get(), {
+        Loaded: (cUser) => cUser,
+        NotLoaded: () => undefined,
+      });
+      if (!user) return;
 
       const array = settingsArray ?? Object.keys(config.settings);
-      const newSettings = { ...settings };
 
-      array.forEach((setting) => {
-        let defaultSetting: SettingsConfigProp<T[Extract<keyof T, string>]> | undefined = undefined;
+      stateOb.update((s) => {
+        const news = s.get(config.storagePath);
+        array.forEach((setting) => {
+          let defaultSetting: SettingsConfigProp<T[Extract<keyof T, string>]> | undefined =
+            undefined;
 
-        for (const key in config.settings) {
-          const conf = config.settings[key];
+          for (const key in config.settings) {
+            const conf = config.settings[key];
 
-          if (conf.storageKey === setting) {
-            defaultSetting = conf;
-            break;
+            if (conf.storageKey === setting) {
+              defaultSetting = conf;
+              break;
+            }
           }
-        }
 
-        if (!defaultSetting) return;
+          if (!defaultSetting || !news) return;
 
-        newSettings[setting as keyof T] = defaultSetting.defaultValue;
+          news[setting] = defaultSetting.defaultValue;
+        });
+        return s.set(config.storagePath, news ?? {});
       });
-
-      update(config.storagePath, newSettings);
-
-      await updateDB(newSettings);
-
-      navigate('', { replace: true });
     },
-    [config, update, updateDB, navigate, settings],
+    [config, stateOb, loadableUser],
   );
 
   const updateSettings = useCallback(
-    async (updates: Settings, shouldPush = false) => {
-      if (!settings) return;
-
-      const newSettings = { ...settings, ...updates };
-
-      if (isEqual(newSettings, settings)) return;
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    (updates: Settings, shouldPush = false) => {
```

Comment:
Since we switched to 'subscribe', not sure how should we pass 'shouldPush'

Reply:
So after looking into it, the reason we have `shouldPush` is because we're storing pagination for tables in settings. Specifically we want to pushState instead of replaceState whenever a table's offset changes.

This is really one of those situations where it's obvious useSettings is the wrong abstraction, but changing the table stuff would be quite difficult here. I think the most obvious fix is to add an equivalent of `shouldPush` to the settings config since it's always tied to a specific setting and check if any of the updated values have that.

That said, Thiago did some stuff using state in https://github.com/determined-ai/determined/pull/5775, but I think that's pretty hard to follow.

Reply:
I wonder why we used `shouldPush` in the first place, why did we treat `tableOffset` differently?

Reply:
Well, for pagination you _do_ want to push state instead of replacing state. That said I just checked and this feature doesn't actually work properly since we're not listening to the back button and acting appropriately. Let's just remove `shouldPush` for now since the only reason for it to exist isn't working properly. I'll make a ticket to get that working in the future but I don't think that should be part of useSettings when we do it.